src/compiler/crystal/semantic/fix_missing_types.cr

```diff
@@ -79,6 +79,12 @@ class Crystal::FixMissingTypes < Crystal::Visitor
     end
   end
 
+  def visit(node : TypeOf)
+    node.expressions.each &.accept self
+    node.type = @program.no_return.metaclass unless node.type?
```

Comment:
It should just be `@program.no_return`. In the spec too.

Reply:
Why?

Reply:
I think correct type for `typeof(raise("").foo)` is `NoReturn:class` because `typeof(raise(""))` is typed as `NoReturn:class`.